p2p/libp2p/netMessenger.go

```diff
@@ -246,6 +246,10 @@ func (netMes *networkMessenger) createPubSub(withMessageSigning bool) error {
 				if errPublish != nil {
 					log.Trace("error sending data", "error", errPublish)
 				}
+			} else {
```

Comment:
I think you can replace lines 244-253 with:
```
if topic == nil {
     log.Warn("writing on a topic that the node did not register on - message dropped", "topic", sendableData.Topic)
     continue
}

errPublish := topic.Publish(netMes.ctx, sendableData.Buff)
if errPublish != nil {
     log.Trace("error sending data", "error", errPublish)
}
```

Reply:
done, refactored, added context closing functionality